bin/migrate-oats-data/submissions/submap/soil_elements.py

```diff
@@ -0,0 +1,129 @@
+def get_soil_rows(rows, cursor):
+    # fetches adjacent land use data, specifically direction, description and type code
+    component_ids = [dict(item)["alr_appl_component_id"] for item in rows]
+    component_ids_string = ', '.join(str(item) for item in component_ids)    
+    soil_rows_query = f"""SELECT * from 
+                        oats.oats_soil_change_elements osc
+                        WHERE osc.alr_appl_component_id in ({component_ids_string})
+                        """
+    cursor.execute(soil_rows_query)
+    soil_rows = cursor.fetchall()
+    return soil_rows
+
+def create_soil_dict(soil_rows):
+    # creates dict contailing fill and remove data
+    alr_id = 'alr_appl_component_id'
+    area = 'project_area'
+    desc = 'material_desc'
+    origin_desc = 'material_origin_desc'
+    duration = 'project_duration'
+    volume = 'volume'
+    depth = 'depth'
+    code = 'soil_change_code'
+
+    soil_dict = {}
+    for row in soil_rows:
+        app_component_id = row[alr_id]
+        if app_component_id in soil_dict:
+
+            if row[code] == 'REMOVE':
+                # if soil_dict[app_component_id]['RMV'] == True:
+                if 'RMV' in soil_dict.get(app_component_id, {}):
+                        print('ignored element_id:',row['soil_change_element_id'])
+                else:
+                    soil_dict[app_component_id]['remove_type'] = row[desc]
+                    soil_dict[app_component_id]['remove_origin'] = row[origin_desc]
+                    soil_dict[app_component_id]['max_remove_depth'] = row[depth]
+                    soil_dict[app_component_id]['total_remove'] = row[volume]
+                    soil_dict[app_component_id]['remove_duration'] = row[duration]
+                    soil_dict[app_component_id]['remove_area'] = row[area]
+                    if 'import_fill' not in soil_dict.get(app_component_id, {}):
+                        soil_dict[app_component_id]['import_fill'] = False                    
+
+                soil_dict[app_component_id]['RMV'] = True
```

Comment:
Introduce an enum with meaningful keys for the magic strings: 'RMV', 'ADD', 'REMOVE', 'ADD'